frame/composable-traits/src/staking/mod.rs

```diff
@@ -102,7 +109,58 @@ pub struct RewardConfig<AssetId, Balance> {
 
 	/// The rewarding rate that increases the pool `total_reward`
 	/// at a given time.
-	pub reward_rate: Perbill,
+	pub reward_rate: RewardRate<Balance>,
+}
+
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo)]
+pub struct RewardRate<Balance> {
+	/// The period that the rewards are handed out in, in seconds.
+	// REVIEW(benluelo): NonZeroU64?
+	pub period: u64,
+	/// The amount that is rewarded each period.
+	pub amount: Balance,
+}
+
+pub struct BoundedU64<const MIN: u64 = 0, const MAX: u64 = { u64::MAX }> {
+	// _marker: PhantomData<(MIN, MAX)>,
+}
+
+// TODO: TESTS!!!
+impl<const MIN: u64, const MAX: u64> Validate<u64, BoundedU64<MIN, MAX>> for BoundedU64<MIN, MAX> {
+	fn validate(input: u64) -> Result<u64, &'static str> {
+		if input > MAX {
+			Err("value too big")
+		} else if input < MIN {
+			Err("value too small")
+		} else {
+			Ok(input)
+		}
+	}
+}
+
+// TODO: TESTS!!!
+// TODO(benluelo): uom?
+impl<Balance> RewardRate<Balance> {
+	pub fn per_second(
+		amount: Balance,
+		seconds: Validated<u64, BoundedU64<1, { u64::MAX }>>,
```

Comment:
The idea was per second**s**, but I see how that could be confusing. Perhaps separate methods? These were somewhat of an experiment, I think it would be fine to just struct literal syntax instead of constructors